### PR TITLE
Result processor implementation in go

### PR DESCRIPTION
**Pull request by samparent97**

### CODE REVIEW

**Comment by BlakeFreer** [comment]
Merging this branch will include the code from #89 since it points to a commit in history of this branch.

**Comment by samparent97** [comment]
> Merging this branch will include the code from #89 since it points to a commit in history of this branch.

Yep, I rebased it on top of that PR. So we will need to get that one in first

File: results/htmlreportgenerator.go [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+	default:
+		return "", fmt.Errorf("unknown ComparisonOperator: %v", tag.CompOp)
+	}
+}
+
+const htmlTemplate = `
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
```
Comment (TylerStAmour) [inline results/htmlreportgenerator.go:125]: Could we move this to a .html file and just read it instead?

File: results/resultaccumulator.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	}
+
+	var existingTests []map[string]any
+	err = json.Unmarshal(data, &existingTests)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal historic tests")
+	}
+
+	return existingTests, nil
+}
```
Comment (TylerStAmour) [inline results/resultaccumulator.go:187]: Does historicsTestsFP just point to a JSON of all previous local tests?

**Review by TylerStAmour — DISMISSED** [review]
Lgtm, no major comments

File: results/htmlreportgenerator.go (hunk 1 above)
Comment (BlakeFreer) [reply results/htmlreportgenerator.go:125]: Made a separate issue for this, see #95

File: results/resultaccumulator.go (hunk 2 above)
Comment (samparent97) [reply results/resultaccumulator.go:187]: yes, its how the old github pages system used to woek

**Review by TylerStAmour — APPROVED** [review]

**Pull request merged**